src/scanner.c

```diff
@@ -44,6 +44,30 @@ static bool scan_block_comment(TSLexer *lexer, bool allow_eof) {
   return false;
 }
 
+static bool scan_doc_comment(TSLexer *lexer) {
+  for(int stack=0;;){
```

Comment:
Is stack allowing us to do infinite lookahead?
Should we put a sensible hard limit on it?

Reply:
I'll remove the stack, since nesting is not allowed by compiler and does not make sense. This will however turn into `while(true)` since we need to scan the entire doc comment (or go until the end of file) not sure if there should be some sort of bound then, as the bound will just be the end of the file, so for large file with unclosed doc comment, the parsing might be slow.